cogs/challenges.py

```diff
@@ -35,42 +35,65 @@ async def on_raw_reaction_add(self, payload):  # Participant role.
+
+            if message.channel.id == 680851820587122700:  # weekly 1
                 submitted = self.bot.guild.get_role(687417501931536478)
                 submission_channel = self.bot.guild.get_channel(729453161885990924)
-            
+
             else:  # weekly 2
                 submitted = self.bot.guild.get_role(715676464573317220)
                 submission_channel = self.bot.guild.get_channel(729453201081761862)
 
             if submitted not in message.author.roles:
                 await message.delete()
                 attach = message.attachments and message.attachments[0]
-                
+
                 if not attach:
-                    msg = f"{message.author.mention} make sure to __upload a " \
-                          f"python file__ that only includes the code required " \
-                          f"for the challenge!"
+                    msg = (
+                        f"{message.author.mention} make sure to __upload a "
+                        f"file__ that only includes the code required "
+                        f"for the challenge!"
+                    )
                     return await message.channel.send(msg, delete_after=10.0)
-                
-                content = "```py\n" + (await attach.read()).decode("u8").replace("`", "\u200b`") + "```"
+
+                filetype = attach.filename.split(".")[-1]
+
+                if len(filetype) > 4 or len(filetype) == 0:  # Most filetypes are 2-3 chars, 4 just to be safe
+                    return await message.channel.send(
+                        "Please upload a file with a __non-empty file extension shorter than 5 chars__!",
+                        delete_after=10.0,
+                    )
+
+                content = (
+                    f"```{filetype}\n"
+                    + (await attach.read()).decode("u8").replace("`", "\u200b`")
+                    + "```"
+                )
                 if len(content) > 2040:
-                    msg = f"{message.author.mention} attachment can't be __more" \
-                          f" than 2040 characters__."
+                    msg = (
+                        f"{message.author.mention} attachment can't be __more"
+                        f" than 2040 characters__."
+                    )
                     return await message.channel.send(msg, delete_after=10.0)
 
                 await message.author.add_roles(submitted)
                 embed = discord.Embed(description=content, color=0x36393E)
-                embed.set_author(name=str(message.author), icon_url=message.author.avatar_url)
-                embed.set_footer(text=f'#ID: {message.author.id} • {len(content)-9} chars')
+                embed.set_author(
+                    name=str(message.author), icon_url=message.author.avatar_url
+                )
+                embed.set_footer(
+                    text=f"#ID: {message.author.id} • {len(content)-9} chars • Language: {filetype}"
```

Comment:
Here you will be able to do `len(code)`

Reply:
Idt thats needed :eyes: the tester will calculate the code length after substituting `\u200b` with `

Reply:
Yeah but it would be better if the correct length was in the embed footer, and replacing the `` ` `` with `` `\u200b`` should be done only in the `content` not in the `code` variable

Reply:
ok